packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -1744,3 +1744,167 @@ export class Addr extends Op {
     stack.push(addr.publicKey);
   }
 }
+
+// get balance for the requested account specified
+// by Txn.Accounts[A] in microalgos. A is specified as an account
+// index in the Accounts field of the ApplicationCall transaction,
+// zero index means the sender
+// [...stack, bigint]
+export class Balance extends Op {
+  readonly interpreter: Interpreter;
+
+  /**
+   * Description: Asserts if arguments length is zero
+   * @param args Expected arguments: [] // none
+   * @param line line number in TEAL file
+   * @param interpreter Interpreter Object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    super();
+    this.interpreter = interpreter;
+
+    assertLen(args.length, 0, line);
+  };
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, 1);
+    let accountIndex = this.assertBigInt(stack.pop());
+    let bal = 0;
+
+    if (accountIndex === BigInt("0")) {
+      const sender = convertToString(this.interpreter.runtime.ctx.tx.snd);
+      const val = this.interpreter.runtime.ctx.state.accounts.get(sender);
+      if (val === undefined) {
+        throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST);
+      }
+      bal = val.amount;
+    } else {
+      this.checkIndexBound(Number(--accountIndex), this.interpreter.runtime.ctx.tx.apat);
+
+      const buf = this.interpreter.runtime.ctx.tx.apat[Number(accountIndex)];
+      const res = this.interpreter.runtime.ctx.state.accounts.get(convertToString(buf));
+
+      if (res === undefined) {
+        throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST);
+      }
+
+      bal = res.amount;
+    }
+
+    stack.push(BigInt(bal));
+  }
+}
+
+// For Account A, Asset B (txn.accounts[A]) pushes to the
+// stack [...stack, 0] if account has no B holding,
```

Comment:
```suggestion
// push to stack [...stack, 0] if account has no B holding,
```

Reply:
Please update the other comments accordingly

Reply:
ok